lib/cog/command/pipeline/initializer.ex

```diff
@@ -68,4 +83,120 @@ defmodule Cog.Command.Pipeline.Initializer do
+                             "email_address" => sender["email"],
+                             "password" => Passwords.generate_password(12)}) do
+              {:ok, user} ->
+                case ChatHandles.set_handle(user, request["adapter"], sender["handle"]) do
+                  {:ok, _} ->
+                    alert_self_registration_success(user, request, state)
+                    :ok
+                  error ->
+                    Logger.error("Failed to auto-register user '#{sender["handle"]}': #{inspect error}")
+                    alert_self_registration_failed(request, state)
+                    error
+                end
+              error ->
+                Logger.error("Failed to auto-register user '#{sender["handle"]}': #{inspect error}")
+                alert_self_registration_failed(request, state)
+                :error
+            end
+          _ ->
+            Logger.error("Failed to auto-register user '#{sender["handle"]}'. No suitable username was found.")
+            alert_self_registration_failed(request, state)
+            :error
+        end
+    end
+  end
+  defp self_register_user(_, _, _) do
+    :ok
+  end
+
+  defp alert_self_registration_success(user, request, state) do
+    {:ok, message} = self_registration_success_message(user, request)
+    publish_response(message, request["room"], request["adapter"], state)
+  end
+  defp alert_self_registration_failed(request, state) do
+    {:ok, message} = self_registration_failed_message(request)
+    publish_response(message, request["room"], request["adapter"], state)
+  end
+
+  defp publish_response(message, room, adapter, state) do
+    response = %{response: message,
+                 room: room}
+    # Remember, it might not be a *chat* adapter we're responding to
+    {:ok, adapter_mod} = Cog.adapter_module(adapter)
+
+    reply_topic = adapter_mod.reply_topic
+
+    # Slack has a limit of 16kb (https://api.slack.com/rtm#limits),
+    # while HipChat has 10,000 characters
+    # (https://developer.atlassian.com/hipchat/guide/sending-messages). Eventually,
+    # over-long messages will need adapter-specific customization
+    # (breaking into multiple messages, using alternative formats
+    # (e.g. Slack attachments), etc.). For now, though, we can at the
+    # very least tell Carrier to emit a warning when we get a message
+    # that looks like it is potentially too big.
+    Connection.publish(state.mq_conn, response, routed_by: reply_topic, threshold: 10000)
+  end
+
+  defp self_registration_success_message(user, request) do
+    adapter = String.to_existing_atom(request["module"])
+    handle = request["sender"]["handle"]
+
+    context = %{
+      handle: handle,
+      first_name: request["sender"]["first_name"],
+      username: user.username,
+      mention_name: adapter.mention_name(handle),
+      display_name: adapter.display_name(),
+    }
+
+    Template.render("any", "self_registration_success", context)
```

Comment:
Why not use the specific adapter name here instead of `"any"`? We'll fallback to `"any"` now anyway, but that leaves open the possibility of adapter-specific templates at a later date.